foreshadow/preparer.py

```diff
@@ -72,66 +70,90 @@ def __init__(
         cleaner_kwargs=None,
         intent_kwargs=None,
         summarizer_kwargs=None,
-        engineerer_kwargs=None,
+        # engineerer_kwargs=None,
         preprocessor_kwargs=None,
-        reducer_kwargs=None,
+        # reducer_kwargs=None,
         exporter_kwargs=None,
         problem_type=None,
         y_var=None,
         **kwargs
     ):
-        flattener_kwargs = _none_to_dict(
+        self.flattener_kwargs = _none_to_dict(
             "flattener_kwargs", flattener_kwargs, cache_manager
         )
-        cleaner_kwargs_ = _none_to_dict(
+        self.cleaner_kwargs = _none_to_dict(
             "cleaner_kwargs", cleaner_kwargs, cache_manager
         )
-        intent_kwargs_ = _none_to_dict(
+        self.intent_kwargs = _none_to_dict(
             "intent_kwargs", intent_kwargs, cache_manager
         )
-        summarizer_kwargs_ = _none_to_dict(
+        self.summarizer_kwargs = _none_to_dict(
             "summarizer_kwargs", summarizer_kwargs, cache_manager
         )
         # # engineerer_kwargs_ = _none_to_dict(
         # #     "engineerer_kwargs", engineerer_kwargs, cache_manager
         # # )
-        preprocessor_kwargs_ = _none_to_dict(
+        self.preprocessor_kwargs = _none_to_dict(
             "preprocessor_kwargs", preprocessor_kwargs, cache_manager
         )
         # # reducer_kwargs_ = _none_to_dict(
         # #     "reducer_kwargs", reducer_kwargs, cache_manager
         # # )
-        exporter_kwargs_ = _none_to_dict(
+        self.exporter_kwargs = _none_to_dict(
             "exporter_kwargs", exporter_kwargs, cache_manager
         )
+        self.y_var = y_var
+        self.problem_type = problem_type
```

Comment:
Why do we change these variables to attributes of the class?

Reply:
This is to suppress a warning complaining that the kwargs need to be part of the instance. That's the protocol of the pipelines.